File: java/137.java
class Solution {
    public int singleNumber(int[] nums) {
        int ans = 0;
        for (int i = 0; i < 32; i++) {
            int count = 0;
            for (int j: nums) {
                count += (j >> i) & 1;
            }

            if (count % 3 != 0) {
                ans |= (1 << i);
            }
        }

        return ans;
    }
}

File: kt/137.kt
class Solution {
    fun singleNumber(nums: IntArray): Int {
        var ans = 0
        (0 until 32).forEach { i ->
            var count = 0
            nums.forEach { num ->
                if (num and (1 shl i) != 0) {
                    count++
                }
            }
            if (count % 3 != 0) {
                ans = ans or (1 shl i)
            }
        }
        return ans
    }
}